src/main/java/io/leonis/zosma/ipc/protobuf/vision/GoalAdapter.java

```diff
@@ -0,0 +1,29 @@
+package io.leonis.zosma.ipc.protobuf.vision;
+
+import static io.leonis.zosma.game.data.Allegiance.*;
+import static io.leonis.zosma.game.data.FieldHalf.*;
+
+import io.leonis.zosma.game.data.*;
+import io.reactivex.functions.BiFunction;
+import org.robocup.ssl.Geometry.GeometryData;
+
+/**
+ * The Class GoalAdapter.
+ *
+ * Adapts a {@link GeometryData} into one {@link Goal} per allegiance.
+ *
+ * @author Jeroen de Jong
+ */
+public class GoalAdapter implements BiFunction<GeometryData, Allegiance, AllegianceTuple<Goal>> {
+  private final GoalDimensionAdapter goalDimensionAdapter = new GoalDimensionAdapter();
+
+  @Override
+  public AllegianceTuple<Goal> apply(final GeometryData geometry, final Allegiance positiveTeamAllegiance) {
```

Comment:
This should be refactored to use `AllegianceTuple<Team>` just like `PlayersAdapter`